contents/blog/the-state-of-plugins.md

```diff
@@ -0,0 +1,81 @@
+hideAnchor: true
+Author: yakko-majuri
+featuredImage: ../images/blog/running-content.png
+featuredImageType: full
+---
+
+The plugin server has become an integral part of PostHog, as it is responsible for event ingestion. However, it isn’t called “ingestion server” because it was originally designed to run plugins. So, how are those going?
+
+I used a combination of PostHog, Metabase, Python scripts, and Google Sheets to compile, analyze, and display the data you see here to answer that question as transparently as possible. 
+
+## TL;DR
+
+- Most teams using PostHog use plugins (if GeoIP is included)
+- Relatively few Cloud teams use plugins (if GeoIP is excluded)
+- Self-hosted users are way more likely to use plugins than Cloud users
+- Export plugins are key for our larger customers (particularly self-hosted ones)
+- Simple plugins covering basic functionality that PostHog lacks are quite popular
+- The plugins ecosystem still needs a lot of work 
+
+## Why we like plugins
+
+Fun fact: the average length for the top six most popular plugins on PostHog is only 57 lines of code. Some, such as the Timestamp Parser, are only 16 lines long! The other thing they have in common is that they cover basic functionality which we haven’t built natively into PostHog yet. 
+
+This is a deliberate decision. It’s true that we could spend engineering time to build these features, such as filtering nested properties, but plugins enable us to quickly solve the same challenge without the resource overhead. Then, once we fully understand the challenge, we can create a better solution — some plugins are literally the duct tape version of full fledged features we’re building at the moment. 
+
+Plugins also enable us to build useful functionality in a matter of hours. The Twitter Followers plugin hits an API and creates an event once a day. It would never make sense to build this natively, but it took just 30 minutes to make it as a plugin which is used by 30+ users.
+
+## About the GeoIP plugin
+
+Despite feeling like a core part of the product, PostHog’s geolocation capabilities are actually handled by a plugin. This is a widely used feature, and the plugin is enabled for everyone by default, meaning it has thousands of users, both on self-hosted instances and Cloud. 
+
+As a result, it has been removed from all the graphs and tables you’ll be seeing in this document, so we can focus on plugins users have deliberately enabled.
+
+## Plugins on PostHog Cloud
+
+PostHog Cloud users have a selection of whitelisted plugins that they can install from [our plugin library](https://posthog.com/plugins). Arbitrary plugins are not allowed, so if users want to install their own plugins, they must first submit a PR to our [Plugin repo](https://github.com/PostHog/plugin-repository) for review.
+
+Out of the more than 4,000 teams we have on PostHog Cloud, 50.3% use plugins (including GeoIP).
+
+However, only 4% of teams use a plugin that is not GeoIP. This is quite low.
```

Comment:
Is it worth mentioning:

- This is a conservative metric including inactive and churned teams
- Through doing this report I found that quite a few plugins weren't available to everyone on Cloud

Fine if not

Reply:
I'm happy to include it, but I trimmed a few details like that out just for conservation of relevancy. We don't revisit those caveats later in the piece, so it didn't seem like they add too much value. That said, if you want them in then we can accomodate.

Reply:
I think this makes sense to add as it helps demonstrate that plugins are valuable.

```suggestion
However, only 4% of teams use a plugin that is not GeoIP. This is quite low. That said, many plugins weren't available to everyone on cloud until recently, so this is likely to be under-reporting the reality of how many teams would use them.
```